scene/animation/animation_player.cpp

```diff
@@ -1152,6 +1152,8 @@ void AnimationPlayer::play(const StringName &p_name, float p_custom_blend, float
 
 	if (String(name) == "")
 		name = playback.assigned;
+	else
+		playback.current.pos = 0;
```

Comment:
This is problematic, as it would reset the animation if you call `play()` with the name of the currently assigned animation. IMO it should only force reset if `name != playback.assigned`.

I also think this shouldn't be changed here, but below on the `c` variable, as done until now. (See other comment.)